content/posts/annoucing-slight-v0_3.md

```diff
@@ -0,0 +1,226 @@
+
+A publisher service can target the `pub` resource to send messages to a topic. Similarly, a subscriber service can choose to subscribe to a topic and then pull the broker to receive messages. This way, the publisher and subscriber services can communicate with each other through the broker, without needing to know each other's details.
+
+Below is an example in Rust showing how you can write a simple publisher service that compiles to Wasm. In this example, the service sends a message to a topic named "rust" and “global-chat” with the messages:
+
+```rust
+use messaging::*;
+wit_bindgen_rust::import!("../../wit/messaging.wit");
+wit_error_rs::impl_error!(messaging::MessagingError);
+
+fn main() -> Result<()> {
+    let ps = Pub::open("my-messaging")?;
+    for i in 0..3 {
+        println!("sending messages");
+        ps.publish(format!("rust-value-{}", i).as_bytes(), "rust")?;
+        ps.publish(format!("gc-value-{}", i).as_bytes(), "global-chat")?;
+        thread::sleep(Duration::from_secs(3));
+    }
+    Ok(())
+}
+```
+
+And an example showing how a subscriber service could be authored:
+
+```rust
+let ps = Sub::open("my-messaging")?;
+let sub_tok = ps.subscribe("rust")?;
+let sub_tok1 = ps.subscribe("global-chat")?;
+
+for _ in 0..3 {
+    loop {
+        let msg = ps.receive(&sub_tok)?;
+        if !msg.is_empty() {
+            println!("received message from topic 'rust'> value: {:?}", String::from_utf8(msg));
+            break;
+        }
+    }
+
+    loop {
+        let msg = ps.receive(&sub_tok1)?;
+        if !msg.is_empty() {
+            println!("received message from topic 'global-chat'> value: {:?}", String::from_utf8(msg));
+            break;
+        }
+    }
+}
+```
+
+One limitation of the current messaging interface and implementation is the lack of support for streaming events. This can be problematic in situations where the receive function is a blocking call that tries to pull the message broker for a new message. If the broker does not have any new messages, the receive function will be blocked, which can cause delays in message delivery. Additionally, the lack of streaming support can limit the scalability of the system, as the blocking call can cause bottlenecks in high-traffic scenarios. To address this limitation, we will be adding `time-to-expire` to the receive function in the near future and will be collaborating with the Wasm Component Model community to land `streams` feature in WIT.
+
+### Using the SQL Capability
+
+The latest release from Slight adds an exciting new capability to its portfolio of capabilities. The newly added SpiderLightning functionality enables WebAssembly programs to safely and generically interact with SQL databases. This capability is not only highly versatile, but also secure, as it includes functions for querying and modifying data using prepared statements, as well as handling errors in a way that ensures data integrity.
```

Comment:
_safety_ is in the eye of the beholder. If you can use a sql query, you can do slq injection, which is NOT safe. :-)

So I'd say "to easily and generically" here.